pkg/gui/custom_commands.go

```diff
@@ -151,10 +156,81 @@ func (gui *Gui) handleCustomCommandKeybinding(customCommand config.CustomCommand
 						return gui.surfaceError(err)
 					}
 
+					return gui.createMenu(title, menuItems, createMenuOptions{showCancel: true})
+				}
+			case "menuFromCommand":
+				f = func() error {
+					// Collect cmd to run from config
+					cmdStr, err := gui.resolveTemplate(prompt.Command, promptResponses)
+					if err != nil {
+						return gui.surfaceError(err)
+					}
+
+					// Collect Filter regexp
+					filter, err := gui.resolveTemplate(prompt.Filter, promptResponses)
+					if err != nil {
+						return gui.surfaceError(err)
+					}
+					reg, err := regexp.Compile(filter)
+					if err != nil {
+						return gui.surfaceError(errors.New("unable to parse filter regex, error: " + err.Error()))
+					}
+
+					// Run and save output
+					message, err := gui.GitCommand.RunCommandWithOutput(cmdStr)
+					if err != nil {
+						return gui.surfaceError(err)
+					}
+
+					// Need to make a menu out of what the cmd has displayed
+					candidates := []string{}
+					temp := template.Must(template.New("format").Parse(prompt.Format))
+					for _, str := range strings.Split(string(message), "\n") {
+						if str == "" {
+							continue
+						}
+						buff := bytes.NewBuffer(nil)
+						groupNames := reg.SubexpNames()
+						tmplData := map[string]string{}
+						for matchNum, match := range reg.FindAllStringSubmatch(str, -1) {
+							if len(match) > 0 {
```

Comment:
I doubt if we want loop over every match, from the programming perspective this is an obvious set but i think users expect the program to only to use the first match.
This complicates the `match_..._...` and i think it's not directly clear to many users what the second argument is as they probably do not expect to match multiple times.  
I would say the extra complexity and code outweighs the advantages.

Reply:
Hmm! After giving it a thought I couldn't find a suitable use case for multiple matches so I reverted back to getting only the first match 😄